internal/buffer/save.go

```diff
@@ -150,6 +150,32 @@ func (b *Buffer) saveToFile(filename string, withSudo bool, autoSave bool) error
 	// Removes any tilde and replaces with the absolute path to home
 	absFilename, _ := util.ReplaceHome(filename)
 
+	// Resolve possible symlinks to their absolute path
+	for {
+		// Unfortunately only Stat() checks for recursion
```

Comment:
Not only Stat() "checks for recursion". Any syscall that needs to do anything with the file itself, not with the symlink to it, returns this ELOOP errno if there is a symlink loop, because in this case there is no actual file for it to do anything with, there are only symlinks pointing to each other, not to an actual file.

In particular, `lstat()`, unlike `stat()`, does not return an error in this case, since it deals with the symlink itself, not with the file it points to.

So a correct comment here would be like: "A trick to let the OS do symlink loop detection for us, since we are lazy to do it ourselves"